jax/experimental/general_map.py

```diff
@@ -125,3 +151,19 @@ def sched_fun_wrapper(*args):
 
 gmap_p = core.MapPrimitive('gmap')
 gmap_p.def_impl(gmap_impl)
+
+
+def subst_axis_names(jaxpr, replaced_name, axis_names):
+  eqns = [subst_eqn_axis_names(eqn, replaced_name, axis_names) for eqn in jaxpr.eqns]
+  return core.Jaxpr(jaxpr.constvars, jaxpr.invars, jaxpr.outvars, eqns)
+
+def subst_eqn_axis_names(eqn, replaced_name, axis_names):
+  if isinstance(eqn.primitive, (core.CallPrimitive, core.MapPrimitive)):
+    if eqn.params.get('axis_name', None) == replaced_name:  # Check for shadowing
+      return eqn
+    new_call_jaxpr = subst_axis_names(eqn.params['call_jaxpr'], replaced_name, axis_names)
+    return eqn._replace(params=dict(eqn.params, call_jaxpr=new_call_jaxpr))
```

Comment:
Is `_replace` defined anywhere?

Reply:
https://docs.python.org/3.8/library/collections.html#collections.somenamedtuple._replace